src/envoy/server/manager/log_event.py

```diff
@@ -0,0 +1,76 @@
+import logging
+from datetime import datetime
+
+from envoy_schema.server.schema.sep2.log_events import LogEvent, LogEventList
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from envoy.server.crud.end_device import select_single_site_with_site_id
+from envoy.server.crud.log_event import count_site_log_events, select_log_event_for_scope, select_site_log_events
+from envoy.server.exception import NotFoundError
+from envoy.server.mapper.sep2.log_event import LogEventListMapper, LogEventMapper
+from envoy.server.request_scope import DeviceOrAggregatorRequestScope, SiteRequestScope
+
+logger = logging.getLogger(__name__)
+
+
+class LogEventManager:
+
+    @staticmethod
+    async def fetch_log_event_for_scope(
+        session: AsyncSession,
+        scope: DeviceOrAggregatorRequestScope,
+        log_event_id: int,
+    ) -> LogEvent:
+        """Fetches a LogEvent by id for a specific scope. Failure to find the response will raise a NotFoundError."""
+        site_log_event = await select_log_event_for_scope(session, scope.aggregator_id, scope.site_id, log_event_id)
+        if site_log_event is None:
+            raise NotFoundError(f"LogEvent {log_event_id} either doesn't exist or is inaccessible in this scope")
+        return LogEventMapper.map_to_log_event(scope, site_log_event)
+
+    @staticmethod
+    async def fetch_log_event_list_for_scope(
```

Comment:
Enddevice has a separate manager for individual vs. list entities. Either approach is fine, just need to make it consistent across all managers